### ISSUE TITLE
Move insecure links from "valid" to "invalid"

### ISSUE THREAD

**Issue by sascha11110** [digitalfabrik/integreat-cms#68]
**Updated with all comments from below on March 14th 2025**

### Motivation

Currently we label some links as insecure, for example if they use http instead of https. You can identify insecure links by the "locked" icon which means that the SSL certificate could not be validated.

### Proposed Solution

To be in line with the best practices in web, it would be best to move these insecure links from the "valid" section in our link checker to "invalid". 

### Alternatives

None

### User Story

As a user of the CMS I want to see insecure links in the category of "invalid" so that I immediately know which links need some form of work 

### Additional Context

For documentation the inital report of this issue
<details>
Many users include images or other media from sites that do not use https. As a result, our website is marked as "non-secure" in the browser and in addition, this is a fairly high security risk.

~Since, from a practical point of view, we cannot completely prohibit users from embedding such content, we should take various soft measures that force users to https.~

_Update 2021_: Since meanwhile https is a very popular standard, it should be safe to completely forbid http content and use a hard enforcement here.

On the one hand, new content is problematic, but so is existing content. Therefore I suggest the following measures:
- The editor points out that the placement of such new content is very critical
- According to #46, such contents should not be packaged
- Pages, events, etc. with such content are marked in the corresponding list view
- There should be an additional list of all pages, events, etc. that have such content</details>

### Design Requirements

None

### Related Issues

Unsure

**Comment by svenseeberg** [comment]
I guess we will do this in the nginx configuration?

**Comment by ulliholtgrave** [comment]
We can use the broken link checker to evaluate whether a link can be accessed via HTTPS. This should probably also extended to foreign media sources (e.g. images).

**Comment by svenseeberg** [comment]
I think the best (not-breaking) solution for now would be to add a new list to our link checker that shows all "Insecure Links":

![image](https://user-images.githubusercontent.com/15197000/212468133-6b6e729a-20b4-4eb5-be84-211001406aa5.png)

That means we just need to add a new link in the menu and filter URLs that start with `http://`.

**Comment by timoludwig** [comment]
> I think the best (not-breaking) solution for now would be to add a new list to our link checker that shows all "Insecure Links":

Good idea! I would even argue that this is a useful upstream contribution to https://github.com/DjangoAdminHackers/django-linkcheck. I'll open an issue there and see if I get feedback from the maintainer.

**Comment by osmers** [comment]
@timobrembeck is this something you will work on this milestone or should we move it to 25Q1?

**Comment by JoeyStk** [comment]
@hauf-toni currently insecure links (meaning they use HTTP instead of HTTPS) are marked with a red icon, but still shown in the "valid" tab. It was suggested that we should move this to the "invalid" tab, as they should be exchanged. However there are arguments made for both sides, saying: It should stay in "valid" as they technically work, or to mark them as "invalid" as a call to action should be made. What do you think? Leave it in "valid" and close this issue, or should we rename this issue and make this issue about moving these links to the "invalid" section? Thank you in advance :)

**Comment by hauf-toni** [comment]
From a best practice view, insecure http links should be marked as "invalid" rather than "valid" even if they technically work. This is aligned with web standards: most browsers already flag http pages as "not secure" and often even prevent you from accessing the site before you explicitly approve your action to proceed again. 
Hope that helps – let me know if you want a more elaborate explaination on this or have any more questions.
